app/Http/Controllers/Maps/CustomMapDataController.php

```diff
@@ -139,6 +139,12 @@ public function get(Request $request, CustomMap $map): JsonResponse
 
         foreach ($map->nodes as $node) {
             $nodeid = $node->custom_map_node_id;
+            if ($node->linked_custom_map_id > 0) {
+                $nodes_down = CustomMapNode::where('custom_map_id', $node->linked_custom_map_id)->whereRelation('device', 'status', 0)->get();
```

Comment:
```suggestion
                $nodes_down = CustomMapNode::where('custom_map_id', $node->linked_custom_map_id)->whereRelation('device', 'status', 0)->exists();
```

Reply:
This will be a faster query and return a boolean.

Reply:
updated query to account for things I mentioned, but I did not test it.

Reply:
Tested the changes, works still as expected :)

Reply:
Oopsie, you didn't accept my suggestion :)  Added it in #16526